app/models/dependency.rb

```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
 class Dependency < ApplicationRecord
-  has_many :comments, dependent: :destroy
+  has_many :comments, dependent: :destroy do
+    def published
```

Comment:
This should be defined in the `Comment` model as:
```ruby
# frozen_string_literal: true

class Comment < ApplicationRecord
  belongs_to :dependency

  def self.published
    where(published: true)
  end
end
```

Reply:
For some reason, I tried it and could call `Dependency.first.comments.published` (which is why I used this approach). 
But it's working now, so I guess I had done something wrong earlier :disappointed: 

This solution is indeed far better :100: